### PR TITLE
fix: reply.send supports Uint8Array payload

### PR DESCRIPTION
**Pull request by SgtPooki**
- test: false positive Uint8Array view length
- fix(reply.send): support Uint8Array payloads

This issue fixes https://github.com/fastify/fastify/issues/5118 according to the
agreed upon [comment](https://github.com/fastify/fastify/issues/5118#issuecomment-1778897189).

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/reply.js [hunk 1]
```diff
@@ -164,7 +164,7 @@ Reply.prototype.send = function (payload) {
       if (hasContentType === false) {
         this[kReplyHeaders]['content-type'] = CONTENT_TYPE.OCTET
       }
-      const payloadToSend = Buffer.isBuffer(payload) ? payload : Buffer.from(payload.buffer)
+      const payloadToSend = payload instanceof Uint8Array ? payload : Buffer.from(payload.buffer)
```
Comment (mcollina) [inline lib/reply.js:167]:
```suggestion
      const payloadToSend = Buffer.isBuffer(payload) ? payload : Buffer.from(payload.buffer, payload.byteOffset, payload.byteLength) 
```

I was referring to this form.

File: lib/reply.js [hunk 2]
```diff
@@ -613,7 +613,7 @@ function onSendEnd (reply, payload) {
     return
   }
 
-  if (typeof payload !== 'string' && !Buffer.isBuffer(payload)) {
+  if (typeof payload !== 'string' && !Buffer.isBuffer(payload) && !(payload instanceof Uint8Array)) {
```
Comment (mcollina) [inline lib/reply.js:616]: I don't think this would be needed afterwards

File: lib/reply.js (hunk 1 above)
Comment (SgtPooki) [reply lib/reply.js]: Ah.. good catch. sorry about that.. that does change less things :)

**Comment by SgtPooki** [comment]
FYI, `npm run test && npm run benchmark` succeed locally. https://github.com/fastify/fastify/actions/runs/6656913470/job/18090709228?pr=5124#step:5:10 seems unrelated ¯\\_(ツ)\_/¯

lmk if there's something I can do

File: lib/reply.js [hunk 3]
```diff
@@ -164,7 +164,7 @@ Reply.prototype.send = function (payload) {
       if (hasContentType === false) {
         this[kReplyHeaders]['content-type'] = CONTENT_TYPE.OCTET
       }
-      const payloadToSend = payload instanceof Uint8Array ? payload : Buffer.from(payload.buffer)
+      const payloadToSend = Buffer.isBuffer(payload) ? payload : Buffer.from(payload.buffer, payload.byteOffset, payload.byteLength)
```
Comment (Uzlopak) [inline lib/reply.js]:
```suggestion
      const payloadToSend = Buffer.isBuffer(payload) ? payload : Buffer.from(payload.buffer, payload.byteOffset, payload.byteLength)
```

**Review by metcoder95 — APPROVED** [review]

**Comment by SgtPooki** [comment]
It looks like something else broke lint and other benchmarks now: https://github.com/fastify/fastify/actions/runs/6656913470/job/18097568956?pr=5124. Let me know if you need anything from me.

**Comment by mcollina** [comment]
Looks like lint was a transient error.

**Comment by mcollina** [comment]
(benchmarks seems to be broken)

**Pull request merged**